### ISSUE TITLE
Perhaps Replicache callbacks (`pusher`, `puller`, `mutators`, etc) should propagate errors

### ISSUE THREAD

**Issue by aboodman** [rocicorp/replicache#710]
Right now if most Replicache callback interfaces throw, we catch the error and put it to console.

The rationale is that user code is not typically below us in the stack so if we throw, it's going to be unhandled and end up as a global error.

But some users might prefer that if they are already are handling errors at the global scope. Right now it is hard for users to do anything useful in response to these errors, like put them to Sentry.

**Comment by aboodman** [comment]
I don't think this bug is necessarily "shovel-ready" so removing the fixit. Once we have clear feeling that it would be API improvement to do this, will add back.

**Comment by grgbkr** [comment]
Today developers can wrap these functions with error handling themselves before passing them as options to replicache.  This could work reasonably well for the error reporting (e.g. Sentry) or even displaying Error toasts/dialogs to the user.

I think we can provide a nicer and more obvious api for these use cases, but need to think it through.
